### ISSUE TITLE
[NEW ALGORITHM] Parallel Graph Algorithms

### ISSUE THREAD

**Issue by devxMani** [AlgoGenesis/C#819]
## Issue will be closed if:

> 1) You mention more than one algorithm. You can create a separate issue for each algorithm once the current one is completed.  
> 2) You propose an algorithm that is already present or has been mentioned in a previous issue.  
> 3) You create a new issue without completing your previous issue.

**Note:** These actions will be taken seriously. Failure to follow the guidelines may result in the immediate closure of your issue.

---

### Name:  
```Parallel Graph Algorithms - Graph```  

### About:  
Algorithms optimized for graph processing on large-scale data (such as social networks), which can process millions of nodes and edges in parallel.

---

### Labels:  
```new algorithm, gssoc-ext, hacktoberfest, level1```

---

### Assignees:  
- [ ] Contributor in GSSoC-ext
- [ ] Want to work on it

**Comment by pankaj-bind** [comment]
Do not create unnecessary PR until you complete the previous one.

**Comment by devxMani** [comment]
@pankaj-bind okk 1 PR at a time now can i create new PR for this my old PRs are merged.
